guides/source/active_storage_overview.md

```diff
@@ -473,6 +473,13 @@ message.video.open do |file|
 end
 ```
 
+Analyzing Files
+---------------
+
+Active Storage [analyzes](https://api.rubyonrails.org/classes/ActiveStorage/Blob/Analyzable.html#method-i-analyze) files after they've been uploaded by queuing a job in Active Job. Analyzed files will store additional information in the metadata hash, including `analyzed: true`. You can check whether a blob has been analyzed by calling `analyzed?` on it.
+
+Image analysis provides `width` and `height` attributes. Video analysis provides these, as well as `duration`, `angle`, and `display_aspect_ratio`. Video analysis also requires the [FFmpeg](https://www.ffmpeg.org/) library, which you must include separately.
```

Comment:
For the image analysis, the user will have to install ImageMagick and the `mini_magick` gem, you might want to mention that for the image analysis part.

Reply:
That’ll be image_processing instead, the mini_magick integration is (soft-?)deprecated.

Reply:
Ah I see. Perhaps the documentation for `ImageAnalyzer` should be updated to reflect that? https://github.com/rails/rails/blob/master/activestorage/lib/active_storage/analyzer/image_analyzer.rb#L13

Reply:
Ah yes, PR welcome! Check the Rails 6 release notes or upgrading guides. One of those should have the details.

Reply:
@peterzhu2118 @kaspth Thanks. I've added a [new commit](https://github.com/rails/rails/pull/37751/commits/480b24264261ba18bc8af3f4215f9b61688fbb4f) to include the `image_processing` requirement for analysis. And I've slightly revised the following section on image transformation to simplify / clarify.

Reply:
`image_processing` is used for variants, not analysis. Analysis still requires `mini_magick`.

Reply:
Oh ok, I totally thought we were all in on image_processing. Any particular reason we aren’t using it for analysis? Or just that it hasn’t been update for whatever reason?

Reply:
`image_processing` isn’t for reading image metadata, as far as I know. It only supports transformation.